public/app/plugins/datasource/prometheus/datasource.tsx

```diff
@@ -771,7 +771,7 @@ export class PrometheusDatasource
     return await lastValueFrom(
       getBackendSrv()
         .fetch<BackendDataSourceResponse>({
-          url: '/api/ds/query',
+          url: `/api/ds/query?type=${this.type}&uid=${this.uid}`,
```

Comment:
Just leaving a comment/reminder here, but not specifically applicable to this datasource or these changes.

Something that has been discussed historically but we never found a good solution is if the PanelQueryRunner (the one running the queries) could pass "context" (somehow) that backend_srv could pick up. I'm not sure if your changes are applicable to this, but we do have for Prometheus and Graphite the following thing:
https://github.com/grafana/grafana/blob/9b8e3b96198685c8a1528015f6bce8be3602a11e/public/app/plugins/datasource/prometheus/datasource.tsx#L178-L183

Would be nice if we could populate these kind of things in backend_srv, if available, so that not every single plugin needs to implement it :)

Reply:
I added that to the standard DataSourceWIthBackend. (except for the internal dashboard ID)